code/modules/mob/living/silicon/robot/dogborg/dog_sleeper_vr.dm

```diff
@@ -126,6 +128,9 @@
 		dat += "<font color='red'><B>Current load:</B> [length(contents)] / [garbo.max_item_count] objects.</font><BR>"
 		dat += "<font color='gray'>([list2text(contents,", ")])</font><BR><BR>"
 
+	if(istype(src, /obj/item/device/dogborg/sleeper/compactor/analyzer))
+		dat += "<A href='?src=\ref[src];sync=1'>Sync Files</A><BR>"
```

Comment:
Use a var instead of this istype check, it's less expensive.

If you want to know what you should REALLY do, put some #defines at the top that represent capabilities, and add a var just called `capabilities` and make it a bitfield of capabilities, which will be very fast.